tests/eas/st_hold.py

```diff
@@ -0,0 +1,408 @@
+        cpu_g.set(cpus=cpus, mems=mems)
+
+    @classmethod
+    def setup_schedtune_cgroup(self, cgroup, boost, prefer_idle):
+        st_g = self.schedtune_controller.cgroup(cgroup)
+        st_g.set(boost=boost, prefer_idle=prefer_idle)
+
+    @classmethod
+    def setup_cgroups(self):
+        """
+        When we set up the cgroups in a unified hierarchy, we
+        need to ensure that the cpu controller is minimally
+        configured before we can run tasks inside a group.
+        Copy the root group's mems and cpus parameters into the
+        groups we create just in case.
+        """
+        self.cpuset_controller = self.target.cgroups.controller('cpuset')
+        root_g = self.cpuset_controller.cgroup('/')
+        params = root_g.get()
+        self.schedtune_controller = self.target.cgroups.controller('schedtune')
+        self.setup_schedtune_cgroup(self.boosted_cgroup_name, 100, 0)
+        self.setup_cpuset_cgroup(self.boosted_cgroup_name,
+                                params['cpus'], params['mems'])
+        self.setup_schedtune_cgroup(self.unboosted_cgroup_name, 0, 0)
+        self.setup_cpuset_cgroup(self.unboosted_cgroup_name,
+                                params['cpus'], params['mems'])
+
+    @classmethod
+    def get_task_times(self):
+        """
+        figure out timestamps where task is running
+        defined here as:
+          start_time = first time we switch to the correctly named task
+          end_time   = last time we switch away from the correctly named task
+        """
+        df = self.ftrace_obj.sched_switch.data_frame
+        self.task_start_time = df[df.next_comm == self.task_name].iloc[0].name
+        self.task_end_time = df[df.prev_comm == self.task_name].iloc[-1].name
+
+    @classmethod
+    def avg_cpu_freq(self):
+        """
+        To calculate average frequency, first obtain the area under the
+        series of frequency transitions for the period of interest. We do
+        this by recomputing the index on the frequency series derived from
+        the cpu_frequency trace objects and then using area_under_curve from
+        bart. Once we have the area, we can divide it by the time span to
+        obtain average frequency.
+        """
+        df = self.ftrace_obj.cpu_frequency.data_frame
+        cpu_df = df[df.cpu == self.test_cpu]
+        freq_s = cpu_df.frequency
+        old_index = list(freq_s[self.task_start_time:self.task_end_time].index)
+        # the new index should run from the start to the end time, including
+        # any events which occurred in between
+        new_index = [ self.task_start_time ] + old_index + [ self.task_end_time ]
+        windowed_freq_s = freq_s.reindex(index=new_index, method='pad')
+        # avg_freq is area / time for the window of interest
+        area = area_under_curve(windowed_freq_s, method='rect')
+        return int((area + 0.5) / (self.task_end_time - self.task_start_time))
```

Comment:
Since this code is using division, it should use this statement at the beginning:
from __future__ import division
And then the code needs to be adapted so it works properly. That will ensure the same behavior on Python2 and 3 without modifying the code itself